pkg/api/validation/validation.go

```diff
@@ -372,6 +372,11 @@ func ValidateObjectMetaUpdate(newMeta, oldMeta *api.ObjectMeta, fldPath *field.P
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("resourceVersion"), newMeta.ResourceVersion, "must be specified for an update"))
 	}
 
+	// Generation should be the same or at most incremented by one.
+	if newMeta.Generation != oldMeta.Generation && newMeta.Generation != oldMeta.Generation+1 {
```

Comment:
Callers compare it, they don't set it. Apiserver sets it in spec, controllers set it in status. We should disallow changes to the generation from clients alltogether, but validation runs after PrepareForUpdate (haven't checked but it must, right?). If there's a way to say: user is never allowed to set it, that would be ideal.

The only thing that uses generation today is the rc manager (https://github.com/kubernetes/kubernetes/blob/ee2a0694b649941fc0c3be606746db041b75b91d/pkg/registry/controller/strategy.go#L72). I think the kubelet is going to start using it.